lib/manager/npm/post-update/yarn.ts

```diff
@@ -46,15 +49,38 @@ export async function generateLockFile(
   logger.debug(`Spawning yarn install to create ${lockFileName}`);
   let lockFile = null;
   try {
-    let installYarn = 'npm i -g yarn';
-    const yarnCompatibility = config.compatibility?.yarn;
-    if (validRange(yarnCompatibility)) {
-      installYarn += `@${quote(yarnCompatibility)}`;
+    const configuration = await Configuration.find(
+      npath.toPortablePath(cwd),
+      null,
+      { strict: false }
+    ).catch(() => null);
+    const yarnPath = configuration && configuration.get('yarnPath');
+
+    const yarnVersion =
+      yarnPath && (await exec(`${yarnPath} --version`)).stdout;
+    const isYarn1 = !yarnVersion || semver.getMajor(yarnVersion) === 1;
```

Comment:
There is an other way we could do this: checking the lockfile (if __metadata exists, it's yarn 2). This version prevent installing yarn 1 if we don't need it, but I'm not sure it's the right one ?

Reply:
I would prefer to continue install global yarn (eg install latest if v2 is requested) and call thatwith the new args / env vars